client/blocks/site-icon/index.jsx

```diff
@@ -27,6 +27,7 @@ function SiteIcon( { siteId, site, iconUrl, size, imgSize, isTransientIcon } ) {
 	const classes = classNames( 'site-icon', {
 		'is-blank': ! iconSrc,
 		'is-transient': isTransientIcon,
+		'has-icon': !! iconSrc,
```

Comment:
I reckon we need both this and `is-blank` because we want to apply a background colour in all cases _except_ when there's an icon.